pkg/tests/cross-tests/input_cross_test.go

```diff
@@ -217,3 +217,35 @@ func TestInputsEmptyConfigModeAttrSet(t *testing.T) {
 		SkipCompareRawConfig: true,
 	})
 }
+
+func TestExplicitNilList(t *testing.T) {
+	skipUnlessLinux(t)
+	// This is an explicit null on the tf side:
+	// resource "crossprovider_testres" "example" {
+	//     f0 = null
+	// }
+	// This is different to an unspecified value.
+	// We might not be able to distinguish between null and missing collections.
```

Comment:
Have you dug into this to make this a bit more precise?

I think there are cases where the distinction ends up being quite meaningful, like TF diffs can be sensitive to this distinction. If/when we lose the distinction in translation is a small window for issues to creep in.

Reply:
The one interesting case which we definitely need to test and annotate explicitly is handling blocks. If I understand this right blocks cannot receive nulls in TF (or unknowns) and that's structurally prevented, however in Pulumi the projection of blocks will always be able to be called with nulls and unknowns and we have to compensate for that possibility.

Reply:
The comment here is more about the fact that we might not be able to express nil vs missing in our programs or in the bridge.

We need to do a bit more testing but I strongly suspect that we just don't have a concept of an explicitly nil value vs an unspecified value.

Reply:
Also I'm not sure most languages have a distinction there too - maybe just Typescript?

Reply:
I might be wrong here, I'll double test as the HCL docs do say that an explicit null should be equivalent to an unspecified value.

Perhaps the one exception here is ConfigModeAttr hacks, but that's probably fine.